File: typescript/src/main/vsdk/toolkit/app/options/OptionsGroupRadianceMethod.ts
import { OptionBase } from "../../common/commandLineOptions/OptionBase";
import { OptionGroup } from "../../common/commandLineOptions/OptionGroup";
import { OptionParser } from "../../common/commandLineOptions/OptionParser";
import { TypedOption } from "../../common/commandLineOptions/TypedOption";

export class OptionsGroupRadianceMethod {
  private constructor() {
  }

  public static radianceMethodParseOptions(
    argc: number[],
    argv: string[],
    radianceMethodsStringOut: string[]
  ): void {
    const radianceMethodName = TypedOption.valueRef<string | null>(null);
    const radianceMethodOpt = new TypedOption<string | null>(
      "-radiance-method",
      radianceMethodName,
      1,
      null,
      null
    );
    const radianceOptions: OptionBase[] = [
      TypedOption.REGISTER_OPTION(radianceMethodOpt, 4),
    ];
    const radianceGroups = [
      new OptionGroup("radiance", radianceOptions, 1),
    ];
    OptionParser.parse(argc, argv, radianceGroups, 1);

    if (radianceMethodsStringOut !== null && radianceMethodsStringOut.length > 0) {
      radianceMethodsStringOut[0] = radianceMethodName.get() !== null ? radianceMethodName.get() as string : "";
    }
  }
}

File: cpp/testsuite/ApplicationCases/RenderparkApplication/src/options/OptionsGroupRadianceMethod.cpp
#include <cstring>

#include "vsdk/toolkit/common/commandLineOptions/OptionParser.h"
#include "vsdk/toolkit/common/commandLineOptions/TypedOption.h"
#include "options/OptionsGroupRadianceMethod.h"

void
OptionsGroupRadianceMethod::radianceMethodParseOptions(
    int *argc,
    char **argv,
    char *radianceMethodsStringOut)
{
    char *radianceMethodName = nullptr;
    TypedOption<char *> radianceMethodOpt = {"-radiance-method", &radianceMethodName, 1, nullptr, nullptr};
    OptionBase radianceOptions[] = {
        REGISTER_OPTION(char *, radianceMethodOpt, 4)
    };
    OptionGroup radianceGroups[] = {
        OptionGroup("radiance", radianceOptions, 1)
    };
    OptionParser<OptionBase>::parse(argc, argv, radianceGroups, 1);

    if ( radianceMethodsStringOut != nullptr ) {
        if ( radianceMethodName != nullptr ) {
            strcpy(radianceMethodsStringOut, radianceMethodName);
        } else {
            radianceMethodsStringOut[0] = '\0';
        }
    }
}
